packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -142,14 +142,10 @@ BuildApp() {
     mkdir "${derived_dir}/engine"
     RunCommand cp -r -- "${flutter_podspec}" "${derived_dir}/engine"
     RunCommand cp -r -- "${flutter_framework}" "${derived_dir}/engine"
-    # Make headers, plists, and modulemap files read-only to discourage editing.
-    RunCommand find "${derived_dir}/engine/Flutter.framework" -type f \( -name '*.h' -o -name '*.modulemap' -o -name '*.plist' \) -exec chmod a-w "{}" \;
```

Comment:
Could we prepend a warning message onto the file if there isn't one there already? Like:
```
... -exec cat <(echo "DO NOT EDIT") "{}" \;
```
Or maybe put a message there directly in the engine repo.